client/rest-high-level/src/main/java/org/elasticsearch/client/RollupRequestConverters.java

```diff
@@ -93,6 +94,20 @@ static Request deleteJob(final DeleteRollupJobRequest deleteRollupJobRequest) th
         return request;
     }
 
+    static Request search(final SearchRequest request) throws IOException {
+        if (request.types().length > 0) {
+            /*
```

Comment:
The more I look at this the less I like it. I think the other option is to build a HLRC side request object that uses the server-side agg builders. I'm not sure if that is worth it, but it would be a cleaner API. It wouldn't be exactly the API we want in the end, but it'd be better than this.

Reply:
@hub-cap pointed me to HLRC's `CountRequest` which is pretty similar to what I'm describing. Different because instead of rendering a `SearchSourceBuilder` it we'd just do aggregations and queries and stuff like that.

Reply:
I understand the concern but I find it easier if the request object is the same since you can directly use it for rollup or regular search seamlessly.

Reply:
I see that argument too!

I think in an ideal world the request for search and the request for rollup search would implement the same interface or have a common superclass....

Reply:
@hub-cap, what do you think about merging this like it stands now?

Reply:
can we just placeholder the client side object and test the validation there instead? just extend the existing request? if not then im ok merging as is.

Reply:
I can't extend `SearchRequest` because it is `final`. I'll merge as is then. I *think* it'd be nice to make the client side search request objects for rollup search at the same time as we make them for search. One day, one day....